pkg/pod/pods.go

```diff
@@ -285,74 +275,55 @@ func getPod(uuid *types.UUID) (*pod, error) {
 	if err != nil {
 		return nil, errwrap.Wrap(errors.New("unable to resolve UUID"), err)
 	}
 
-	p, err := getPod(podUUID)
+	p, err := getPod(dataDir, podUUID)
 	if err != nil {
 		return nil, errwrap.Wrap(errors.New("unable to get pod"), err)
 	}
 
 	return p, nil
 }
 
-// path returns the path to the pod according to the current (cached) state.
-func (p *pod) path() string {
-	if p.isEmbryo {
-		return p.embryoPath()
-	} else if p.isPreparing || p.isAbortedPrepare {
-		return p.preparePath()
-	} else if p.isPrepared {
-		return p.preparedPath()
-	} else if p.isExitedGarbage {
-		return p.exitedGarbagePath()
-	} else if p.isGarbage {
-		return p.garbagePath()
-	} else if p.isGone {
-		return "" // TODO(vc): anything better?
-	}
-
-	return p.runPath()
-}
-
 // embryoPath returns the path to the pod where it would be in the embryoDir in its embryonic state.
-func (p *pod) embryoPath() string {
-	return filepath.Join(embryoDir(), p.uuid.String())
+func (p *Pod) embryoPath() string {
+	return filepath.Join(embryoDir(p.dataDir), p.UUID.String())
 }
 
 // preparePath returns the path to the pod where it would be in the prepareDir in its preparing state.
-func (p *pod) preparePath() string {
-	return filepath.Join(prepareDir(), p.uuid.String())
+func (p *Pod) preparePath() string {
+	return filepath.Join(prepareDir(p.dataDir), p.UUID.String())
 }
 
 // preparedPath returns the path to the pod where it would be in the preparedDir.
-func (p *pod) preparedPath() string {
-	return filepath.Join(preparedDir(), p.uuid.String())
+func (p *Pod) preparedPath() string {
+	return filepath.Join(preparedDir(p.dataDir), p.UUID.String())
 }
 
 // runPath returns the path to the pod where it would be in the runDir.
-func (p *pod) runPath() string {
-	return filepath.Join(runDir(), p.uuid.String())
+func (p *Pod) runPath() string {
+	return filepath.Join(runDir(p.dataDir), p.UUID.String())
 }
 
 // exitedGarbagePath returns the path to the pod where it would be in the exitedGarbageDir.
-func (p *pod) exitedGarbagePath() string {
-	return filepath.Join(exitedGarbageDir(), p.uuid.String())
+func (p *Pod) exitedGarbagePath() string {
+	return filepath.Join(exitedGarbageDir(p.dataDir), p.UUID.String())
 }
 
 // garbagePath returns the path to the pod where it would be in the garbageDir.
-func (p *pod) garbagePath() string {
-	return filepath.Join(garbageDir(), p.uuid.String())
+func (p *Pod) garbagePath() string {
+	return filepath.Join(garbageDir(p.dataDir), p.UUID.String())
 }
 
-// xToPrepare transitions a pod from embryo -> preparing, leaves the pod locked in the prepare directory.
+// ToPrepare transitions a pod from embryo -> preparing, leaves the pod locked in the prepare directory.
 // only the creator of the pod (via newPod()) may do this, nobody to race with.
-func (p *pod) xToPreparing() error {
+func (p *Pod) ToPreparing() error {
 	if !p.createdByMe {
 		return fmt.Errorf("bug: only pods created by me may transition to preparing")
```

Comment:
@s-urbaniak This is copied from the old file. I think the "bug" means the error is not expected,